src/commands/login.rs

```diff
@@ -58,9 +63,17 @@ pub fn execute<'a>(_matches: &ArgMatches<'a>) -> Result<(), Error> {
         }
     }
 
-    update_config(&config, &token)?;
-    println!();
-    println!("Stored token in {}", config.get_filename().display());
+    let mut config_to_update = local_config;
+    if matches.is_present("global") {
+        let (global_filename, global_config) = load_global_config_file()?;
+        config_to_update = Config::from_file(global_filename, global_config)?.bind_to_process();
```

Comment:
Is there a better way to get an `Arc` reference here?

Reply:
Which part exactly? You could probably:

- Add a function `Config::global()?` which runs the above two lines 
- Remove bind_to_process, which you probably don't need?

That way you leave functions like `load_global_config_file` internal, and you just expose the `Config` object.